astropy/utils/data.py

```diff
@@ -1311,18 +1315,15 @@ def _get_download_cache_locs():
 
 def _open_shelve(shelffn, withclosing=False):
     """
-    opens a shelf in a way that is py3.x and py2.x compatible.  If
-    `withclosing` is  True, it will be opened with closing, allowing use like:
+    Opens a shelf file.  If `withclosing` is  True, it will be opened with closing,
+    allowing use like:
 
         with _open_shelve('somefile',True) as s:
```

Comment:
Yikes, this cries out for a little class with an `__enter__` and `__exit__` method -- but fine to ignore here!